api/src/main/java/org/infinispan/api/annotations/indexing/Indexed.java

```diff
@@ -37,4 +37,28 @@
     */
    boolean enabled() default true;
 
+   /**
+    * Enables indexes / queries by keys.
+    *
+    * @return The fully qualified name of the type of entities used as keys
+    */
+   String keyEntity() default "";
```

Comment:
We could also have a `Class<?> keyClass` to make it a bit more "typesafe". These would be mutually exclusive

Reply:
The problem is that for the remote entity mapping we want to put here the Protobuf message type and we don't support the embedded case at the moment.